### DISCUSSION TITLE
Can't read the state of the pins

### DISCUSSION THREAD

**Discussion by Silaszur** [Pi4J/pi4j#288] [Q&A]
Hello,

I tried to read and change the state of the pins on my raspberry pi 4 B with a code similar to the following:

```
pi4j = Pi4J.newAutoContext();
DigitalOutput output = null;
var outputConfig3 = DigitalOutput.newConfigBuilder(pi4j)
                    .address(intToPin(pin))
                    .provider("pigpio-digital-output");
output = pi4j.create(outputConfig3);
System.out.println(output.state());
output.on();
output.off();
pi4j.shutdown();
```

And for me only the on and off works. So I can set the states of the pins but not read them out. 
When I read them out the state is "Unknown". And for some reason I could read the state when I changed the config to this
```
var outputConfig3 = DigitalOutput.newConfigBuilder(pi4j)
                    .address(intToPin(pin))
                    .shutdown(DigitalState.HIGH)
                    .provider("pigpio-digital-output");
```

But I don't want to set my pin to High each time that the application closes.
I didn't find anything related to the pin state Unknown, so if anyone got an Idea how I can start to read the pins without using `.shutdown(DigitalState.HIGH)` I would greatly appreciate it.

**Comment by FDelporte** [comment]
Can you please check this example: https://pi4j.com/getting-started/minimal-example-application/

An input needs a different config, for example:

```
var buttonConfig = DigitalInput.newConfigBuilder(pi4j)
      .id("button")
      .name("Press button")
      .address(PIN_BUTTON)
      .pull(PullResistance.PULL_DOWN)
      .debounce(3000L)
      .provider("pigpio-digital-input");
```

**Comment by taartspi** [comment]
Some of this conversation is visible only in email, not in github.  And I as I can't view all your code I will make some comments.   You mentioned setting the pin high before your app exits..  You coded  .shutdown(DigitalState.HIGH), this will ensure your Gpio pin is high after the application exists.      Also, I copied the code from the email July 17 2:43    The DigitalInput create failed until I added the line   pi4j = Pi4J.newAutoContext(); immediately after the call to shutdown().       With this one line addition the code functions as I think you wanted.      But, as I am making assumptions, if you still have questions please post the complete file so we know we are attempting what you want.
